resources/design-document.md

```diff
@@ -0,0 +1,223 @@
+
+U3. As a user, I want to be able to filter through other user's ads.
+
+U4. As a user, I want to be able to change bulletin boards.
+
+U5. As a user, I want to view venues associated with a board.
+
+## 4. Project Scope
+### 4.1 In Scope
+
+* Creating, retrieving, and updating an ad
+* Retrieving all ads a user has created
+* Updating a user's details. 
+
+### 4.2 Out of Scope
+
+* Multiple Boards (broader than Nashville area) - mid stretch
+* Venues as an entity rather than a String - closest stretch?
+* Allow users to apply to ads and view applications to their ads - farthest stretch 
+
+## 5. Proposed Architecture Overview
+
+The MLP will include creating, retrieving, and updating an ad. Included will also be the functionality to update a user's personal details.
+
+API Gateway and Lambda will be used to create these endpoints to satisfy our requirements:
+* `GetAd`
+* `CreateAd`
+* `UpdateAd`
+* `DeleteAd`
+* `GetUser`
+* `CreateUser`
+* `UpdateUser`
+
+Boards, users, ads, locationsm and venues will be stored within DynamoDB in their own respective tables. 
+
+
+## 6. API
+### 6.1 Public Models
+````
+// UserModel
+
+String userId;
+String name;
+String bio;
+List<String> groups;
+List<String> ads; //adId
+````
+
+````
+// AdModel
+
+String adId;
+String name;
+String description;
+Double salary; //double?
+String location; //venue name
+String userId;
+Set<String> tags;
+````
+
+````
+// LocationModel
+
+String locationId;
+String name;
+List<String> venues; //venueId
+````
+
+````
+// VenueModel
+
+String venueId;
+String name; 
+String description;
+String location;
+````
+
+````
+// BoardModel
+
+String boardId;
+String name;
+Set<String> users; //userIds
+Set<String> ads; //adIds
+String location;
+````
+
+### 6.2 Get Ad Endpoint
+* Accepts `GET` requests to `/ads/:adId`
+
+### 6.3 Create Ad Endpoint
+* Accepts `POST` requests to `/ads`
+
+### 6.4 Update Ad Endpoint
+* Accepts `PUT` requests to `/ads/:adId`
+
+### 6.5 Delete Ad Endpoint
+* Accepts `DELETE` requests to `/ads/:adID`
+
+### 6.6 Get User Endpoint
+* Accepts `GET` requests to `/users/:userId`
+
+### 6.7 Create User Endpoint 
+* Accepts `POST` requests to `/users`
+
+### 6.8 Update User Endpoint
+* Accepts `PUT` requests to `/users/:userId`
+
+## 7. Tables
+### 7.1 `boards`
+````
+boardId // partition key, string
+name // string
+usersList // stringSet (userId)
+adsList // stringSet (adId)
+location // stringSet (locationId)
+````
+
+### 7.2 `ads`
+````
+Id // partition key, string
+name // string
+userId // string
+tags // stringSet
+````
+
+### 7.3 `users`
+````
+userId // partition key, string
+name // string
+bio // string
+group // stringList
+adsList // stringSet (adId)
+roles // stringList
+````
+
+### 7.4 `locations`
+````
+locationId // partition key, string
+venuesList // stringSet (venueId)
+````
+
+### 7.5 `venues`
+````
+venueId // partition key, string
+name // string
+description // string
+location // string (locationId)
+````
+
+## 8. Pages
```

Comment:
I like the level of detail and actual sketching you've done here. It'd be helpful to include a brief heading/label for each one. E.g. "Home Page", "Adding an Ad"... etc.